pkg/chartvalues/apiextensions_azure_config_e2e_test.go

```diff
@@ -0,0 +1,215 @@
+			case err == nil && tc.errorMatcher != nil:
+				t.Fatalf("error == nil, want non-nil")
+			case !tc.errorMatcher(err):
+				t.Fatalf("error == %#v, want matching", err)
+			}
+
+			if tc.errorMatcher != nil {
+				return
+			}
+
+			line, difference := rendertest.Diff(values, tc.expectedValues)
+			if line > 0 {
+				t.Fatalf("line == %d, want 0, diff: %s", line, difference)
+			}
+		})
+	}
+}
+
+func Test_NewAPIExtensionsAzureConfigE2E_invalidConfigError(t *testing.T) {
+	testCases := []struct {
+		name         string
+		config       APIExtensionsAzureConfigE2EConfig
+		errorMatcher func(err error) bool
+	}{
+		{
+			name: "case 0: invalid .Azure.CalicoSubnetCIDR",
+			config: newAPIExtensionsAzureConfigE2EConfigFromFilled(func(v *APIExtensionsAzureConfigE2EConfig) {
+				v.Azure.CalicoSubnetCIDR = ""
+			}),
+			errorMatcher: IsInvalidConfig,
+		},
+		{
+			name: "case 1: invalid .Azure.CIDR",
+			config: newAPIExtensionsAzureConfigE2EConfigFromFilled(func(v *APIExtensionsAzureConfigE2EConfig) {
+				v.Azure.CIDR = ""
+			}),
+			errorMatcher: IsInvalidConfig,
+		},
+		{
+			name: "case 2: invalid .Azure.Location",
+			config: newAPIExtensionsAzureConfigE2EConfigFromFilled(func(v *APIExtensionsAzureConfigE2EConfig) {
+				v.Azure.Location = ""
+			}),
+			errorMatcher: IsInvalidConfig,
+		},
+		{
+			name: "case 3: invalid .Azure.MasterSubnetCIDR",
+			config: newAPIExtensionsAzureConfigE2EConfigFromFilled(func(v *APIExtensionsAzureConfigE2EConfig) {
+				v.Azure.MasterSubnetCIDR = ""
+			}),
+			errorMatcher: IsInvalidConfig,
+		},
+		{
+			name: "case 4: invalid .Azure.VPNSubnetCIDR",
+			config: newAPIExtensionsAzureConfigE2EConfigFromFilled(func(v *APIExtensionsAzureConfigE2EConfig) {
+				v.Azure.VPNSubnetCIDR = ""
+			}),
+			errorMatcher: IsInvalidConfig,
+		},
+		{
+			name: "case 5: invalid .Azure.WorkerSubnetCIDR",
+			config: newAPIExtensionsAzureConfigE2EConfigFromFilled(func(v *APIExtensionsAzureConfigE2EConfig) {
+				v.Azure.WorkerSubnetCIDR = ""
+			}),
+			errorMatcher: IsInvalidConfig,
+		},
+		{
+			name: "case 6: invalid .ClusterName",
+			config: newAPIExtensionsAzureConfigE2EConfigFromFilled(func(v *APIExtensionsAzureConfigE2EConfig) {
+				v.ClusterName = ""
+			}),
+			errorMatcher: IsInvalidConfig,
+		},
+		{
+			name: "case 7: invalid .CommonDomain",
+			config: newAPIExtensionsAzureConfigE2EConfigFromFilled(func(v *APIExtensionsAzureConfigE2EConfig) {
+				v.CommonDomain = ""
+			}),
+			errorMatcher: IsInvalidConfig,
+		},
+		{
+			name: "case 8: invalid .CommonDomainResourceGroup",
+			config: newAPIExtensionsAzureConfigE2EConfigFromFilled(func(v *APIExtensionsAzureConfigE2EConfig) {
+				v.CommonDomainResourceGroup = ""
+			}),
+			errorMatcher: IsInvalidConfig,
+		},
+		{
+			name: "case 9: invalid .VersionBundleVersion",
+			config: newAPIExtensionsAzureConfigE2EConfigFromFilled(func(v *APIExtensionsAzureConfigE2EConfig) {
+				v.VersionBundleVersion = ""
+			}),
+			errorMatcher: IsInvalidConfig,
+		},
```

Comment:
Would be good to also have a valid test case, for the sake of completeness.

Reply:
The valid test is above.